### PR TITLE
Documented `~>` version specifier

### PR DESCRIPTION
**Pull request by mverzilli**
Reused the explanation from http://bundler.io/v1.5/gemfile.html

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -146,6 +146,12 @@ A version requirement (String).
 
 Examples: `1.2.3`, `>= 1.0.0` or `~> 2.0`.
 
+Most of the version specifiers, like `>= 1.0.0`, are self-explanatory. 
+The specifier `~>` has a special meaning, best shown by example. 
+`~> 2.0.3` is identical to `>= 2.0.3 and < 2.1`. 
+`~> 2.1` is identical to `>= 2.1 and < 3.0`. 
+`~> 2.2.beta` will match prerelease versions like `2.2.beta.12`.
```
Comment (ysbaddaden) [inline SPEC.md]: I'm not sure about this last one. Better remove it unless we can prove that the non-digit parts in version numbers are working correctly.

File: SPEC.md [hunk 2]
```diff
@@ -146,6 +146,12 @@ A version requirement (String).
 
 Examples: `1.2.3`, `>= 1.0.0` or `~> 2.0`.
 
+Most of the version specifiers, like `>= 1.0.0`, are self-explanatory. 
+The specifier `~>` has a special meaning, best shown by example.
```
Comment (ysbaddaden) [inline SPEC.md]: We use the word `operator` above instead of `specifier`.

Comment (mverzilli) [inline SPEC.md]: Modified.

File: SPEC.md (hunk 1 above)
Comment (mverzilli) [inline SPEC.md]: Agreed. I removed it.

**Pull request merged**